compiler/computation_order/policy_dummy.cc

```diff
@@ -31,4 +31,32 @@ std::vector<Order> DummyPolicy(const Graph& graph) {
     return orders;
 }
 
+std::vector<Order> DummyPolicy2(const Graph& graph) {
+    // discard all the intermediate values in the first run
+    std::vector<Order> orders;
+
+    std::set<Value*> params;
+    for (Value* value : graph.input_values()) {
+        params.insert(value);
+    }
+
+    auto nodes = graph.GetTopologicallySortedNodes();
+    for (auto node : nodes) {
+        orders.emplace_back(Order::kComputeForward, node, nullptr);
```

Comment:
Is this a valid scheduling for all computation graphs? For a graph like
```
v1 = n1(v0)
v2 = n2(v1)
v3 = n3(v1)
v4 = n4(v2, v3)
```
I think the `orders` will be
```
Compute n1
Compute n2
Forget v1
Compute n3
```
at this point, `v1` is not ready? If my concern is right, I think you can use `graph->GetNecessaryNodesAndInputCounts` and do something like
```
zzz = graph->GetNecessaryNodesAndInputCounts()
for (node : nodes) {
  if (!params.count(node) && --zzz[node] == 0)
}
```
I haven't checked this work, though

Reply:
No, dummy2 will output some error for some family of computational graphs.
My intention of adding this policy was to check if retaining intermediate values is working well in two_phase+computation_order mode for simple case.
I do not want to use this policy in practical case.

Reply:
Got it. Could you add `CHECK_EQ(1, value->users().size()) << "Graph with branches is not compatible with this computation order policy";` then?

Reply:
Sure, I would.

Reply:
done